corehq/apps/users/migrations/0031_add_download_reports_permission.py

```diff
@@ -0,0 +1,25 @@
+from django.db import migrations
+
+from corehq.apps.users.models_sql import SQLPermission, SQLUserRole
+from corehq.util.django_migrations import skip_on_fresh_install
+
+
+@skip_on_fresh_install
+def migrate_download_reports_permissions(apps, schema_editor):
+    new_permission_name = 'download_reports'
+    permission, created = SQLPermission.objects.get_or_create(value=new_permission_name)
+    for role in SQLUserRole.objects.all().iterator():
+        if role.permissions.view_reports or bool(role.permissions.view_report_list):
+            role.rolepermission_set.get_or_create(permission_fk=permission, defaults={"allow_all": True})
+            role._migration_do_sync()
```

Comment:
nit: only do sync if the permission was created
```suggestion
            _, created = role.rolepermission_set.get_or_create(permission_fk=permission, defaults={"allow_all": True})
            if created:
                role._migration_do_sync()
```